### PR TITLE
Setup NEHR patient portal internationalization 

### LINKED ISSUE #21: Setup internationalization to support multiple languages.
**Is your feature request related to a problem? Please describe.**
Set up i18n for NEHR client.

**Describe the solution you'd like**

https://react.i18next.com/

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by janithRS**
resolves: https://github.com/lk-gov-health-hiu/project-nhdx/issues/21

Since we will probably want to use Sinhala, English, and Tamil as interface languages, internationalization is needed for the app.

This was done using these packages:

> i18next
> react-i18next 
> i18next-resources-to-backend 
> next-i18n-router

First, react-i18next and its base library i18next were installed.

To assist with loading translations server-side, the i18next-resources-to-backend add-on was used.

Lastly, the next-i18n-router was employed as a lightweight package to implement internationalized routing and locale detection, specifically in App Router projects. It was recommended in Next’s App Router docs, as Next’s built-in internationalized routing feature had been removed in the App Router.

How to use internationalization in the NEHR patient portal

1. Add the relevant namespace strings to /locales ('home.json' includes strings that are displayed on the home page)
2. Update namespaces in namespaces.ts
3. use `useTranslation` hook in child components and use t for setting up the string `{t("importantLinks")}`

Strings used in the app are mentioned in here https://docs.google.com/spreadsheets/d/1EUjABPUtNBeaaqmIA_FgHshLtdl3L1En2CBLrNSJwrA/edit#gid=0

Screenshot 2023-12-17 at 12 16 16 PM
Screenshot 2023-12-17 at 12 16 20 PM

### CODE REVIEW

**Comment by sameeragunarathne** [comment]
@janithRS Shall we link the internationalization doc link in the issue itselt?

File: nehr-patient-portal/i18nConfig.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const i18nConfig = {
+  locales: ['en', 'si', 'ta'],
+  defaultLocale: 'en'
+};
+
+module.exports = i18nConfig;
```
Comment (sameeragunarathne) [inline nehr-patient-portal/i18nConfig.js]: missing EoF

File: nehr-patient-portal/middleware.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import i18nConfig from './i18nConfig';
+
+export function middleware(request) {
+  return i18nRouter(request, i18nConfig);
+}
+
+// applies this middleware only to files in the app directory
+export const config = {
+  matcher: '/((?!api|static|.*\\..*|_next).*)'
+};
```
Comment (sameeragunarathne) [inline nehr-patient-portal/middleware.js]: missing EoF. Check other places as well.

File: nehr-patient-portal/i18nConfig.js (hunk 1 above)
Comment (janithRS) [reply nehr-patient-portal/i18nConfig.js]: Updated with https://github.com/lk-gov-health-hiu/project-nhdx/pull/24/commits/5ba0c50f0b0d13589c036012b77ed68dca087e25

File: nehr-patient-portal/middleware.js (hunk 2 above)
Comment (janithRS) [reply nehr-patient-portal/middleware.js]: Updated with https://github.com/lk-gov-health-hiu/project-nhdx/pull/24/commits/5ba0c50f0b0d13589c036012b77ed68dca087e25

**Pull request merged**